python/cugraph/cugraph/dask/structure/mg_property_graph.py

```diff
@@ -598,7 +598,9 @@ def get_vertex_data(self, vertex_ids=None, types=None, columns=None):
                 # FIXME: invalid columns will result in a KeyError, should a
                 # check be done here and a more PG-specific error raised?
                 df = df[[self.type_col_name] + columns]
-            return df.reset_index()
+            df_out = df.reset_index().persist()
+            df_out.index = df_out.index.astype(df.index.dtype)
+            return df_out
```

Comment:
This will add considerable memory over-head , would it be possible to break down this as a separate PR , we have a big memory overhead anyways ?

Reply:
I could make this a separate PR, but that would block this PR.

Reply:
I'd rather just fix the code.  I think I can safely drop `persist` entirely here.

Reply:
I'm just worried the whole dataframe will get recomputed.  But I will add a check that skips this if there is no change in datatype.

Reply:
`astype` wont  cause dataframe to be recomputed it will just be a task that gets added to the DAG.

Reply:
ok, so the changes I just pushed should resolve this then